db/db_impl/db_impl.cc

```diff
@@ -3135,17 +3135,17 @@ bool DBImpl::GetIntPropertyInternal(ColumnFamilyData* cfd,
     }
   } else {
     SuperVersion* sv = nullptr;
-    if (!is_locked) {
-      sv = GetAndRefSuperVersion(cfd);
-    } else {
-      sv = cfd->GetSuperVersion();
+    if (is_locked) {
+      mutex_.Unlock();
     }
+    sv = GetAndRefSuperVersion(cfd);
 
     bool ret = cfd->internal_stats()->GetIntPropertyOutOfMutex(
         property_info, sv->current, value);
 
-    if (!is_locked) {
-      ReturnAndCleanupSuperVersion(cfd, sv);
+    ReturnAndCleanupSuperVersion(cfd, sv);
+    if (is_locked) {
+      mutex_.Lock();
```

Comment:
Just a question: could lock and unlock for every GetIntProp call have performance issue (assuming most of the GetIntProp calls are fast)? Could the lock/unlock be done at upper level, like `GetAggregatedIntProperty()`?

Reply:
Other props have `need_out_mutex` as `false`, and they do not cause release/acquire of mutex. For `kEstimateTableReadersMem`, its `need_out_mutex` is true, and the operation itself can be expensive if holding mutex all the time.